geometric_kernels/kernels/matern_kernel.py

```diff
@@ -0,0 +1,92 @@
+"""
+A wrapper around different kernels and feature maps that dispatches on space.
+"""
+import lab as B
+from plum import dispatch
+
+from geometric_kernels.kernels import MaternFeatureMapKernel, MaternKarhunenLoeveKernel
+from geometric_kernels.kernels.feature_maps import (
+    deterministic_feature_map_compact,
+    random_phase_feature_map_noncompact,
+)
+from geometric_kernels.spaces import (
+    DiscreteSpectrumSpace,
+    Graph,
+    Mesh,
+    NoncompactSymmetricSpace,
+    Space,
+)
+
+
+@dispatch
+def default_feature_map(space: DiscreteSpectrumSpace, *, num, kernel):
+    return deterministic_feature_map_compact(space, kernel)
+
+
+@dispatch
+def default_feature_map(space: NoncompactSymmetricSpace, *, num, kernel):
+    return random_phase_feature_map_noncompact(space, num)
+
+
+@dispatch
+def default_num(space: Mesh):
+    return min(MaternGeometricKernel._MAX_NUM_EIGENFUNCTIONS, space.num_vertices)
+
+
+@dispatch
+def default_num(space: Graph):
+    return min(MaternGeometricKernel._MAX_NUM_EIGENFUNCTIONS, space.num_vertices)
+
+
+@dispatch
+def default_num(space: DiscreteSpectrumSpace):
+    return MaternGeometricKernel._MAX_NUM_LEVELS
+
+
+@dispatch
+def default_num(space: NoncompactSymmetricSpace):
+    return MaternGeometricKernel._MAX_NUM_RANDOM_PHASES
+
+
+class MaternGeometricKernel:
+    """
+    This class represents a Matérn geometric kernel that "just works".
+
+    Upon creation, this class unpacks into a specific geometric kernel based
+    on the provided Space, and the associated feature map.
+    """
+
+    _MAX_NUM_EIGENFUNCTIONS = 1000
+    _MAX_NUM_LEVELS = 10
```

Comment:
I suggest going for `100`

Reply:
There's a problem with that: https://github.com/vdutor/SphericalHarmonics/issues/11